assertion_test.go

```diff
@@ -477,3 +477,183 @@ func TestAssertion_Strings(t *testing.T) {
 		}
 	})
 }
+
+func TestAssertion_ValidateTraits(t *testing.T) {
+	tests := []struct {
+		testName          string
+		errorContainsText string
+		failure           AssertionFailure
+		traits            fieldTraits
+	}{
+		{
+			testName:          "bad Type",
+			errorContainsText: "AssertionType",
+			failure: AssertionFailure{
+				Type: AssertionType(9999),
+			},
+			traits: fieldTraits{
+				Actual: fieldRequired,
+			},
+		},
+		{
+			testName:          "required Actual",
+			errorContainsText: "should have Actual",
+			failure: AssertionFailure{
+				Actual: nil,
+			},
+			traits: fieldTraits{
+				Actual: fieldRequired,
+			},
+		},
+		{
+			testName:          "denied Actual",
+			errorContainsText: "can't have Actual",
+			failure: AssertionFailure{
+				Actual: &AssertionValue{},
+			},
+			traits: fieldTraits{
+				Actual: fieldDenied,
+			},
+		},
+		{
+			testName:          "required Expected",
+			errorContainsText: "should have Expected",
+			failure: AssertionFailure{
+				Expected: nil,
+			},
+			traits: fieldTraits{
+				Expected: fieldRequired,
+			},
+		},
+		{
+			testName:          "denied Expected",
+			errorContainsText: "can't have Expected",
+			failure: AssertionFailure{
+				Expected: &AssertionValue{},
+			},
+			traits: fieldTraits{
+				Expected: fieldDenied,
+			},
+		},
+		{
+			testName:          "Required value on Expected to be pointer to AssertionRange",
```

Comment:
Here and below, let's simplify test name a bit, e.g. we can call this one just "AssertionRange should not be pointer", and the next one could be just "Min should be non-nil", etc.